### DISCUSSION TITLE
Add some statistics (dates) and pages related to thoses (top recipes, rediscover recipes, etc)

### DISCUSSION THREAD

**Discussion by gabt92** [mealie-recipes/mealie#809] [Feature Request]
**Before submitting this feature request I have**
- [X] Reviewed the documentation and verified that my feature request isn't already a feature.
- [X] Reviewed the existing feature requests and my feature has not been requested.
- [X] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
In my opinion, one thing missing on Mealie is statistics about the recipes. It would be great to know things like, own many times have I consulted each recipes, but more than that, how much time have I cooked it and **when**. Based on those dates, it will then be possible to automatically create things like "cookbook of my most used recipes of 2021" or simply have a page to rediscover recipes that I might have forgotten since I cooked it in 2018, let's say 5 times, and never again after that.

**(Optional): Suggest A Solution**
I'm not technical enough to provide solutions, but I tried to create some mock-ups to help you understand my idea. First thing first, I think that two types of date would be good: 
- When I consulted the recipe
> Simple enough, document the time and date (one max per day) when I consulted the recipe.
- When I cooked the recipe
> This one is, in my opinion, a little bit more complex since everything is shown when we consult the recipe and it's then impossible to really know if the recipe hasn't been cooked only by consulting it. The issue might be partially fixed with the version 1.0 bringing the button 'cook'. ![cook](https://user-images.githubusercontent.com/38195308/142882849-2bf6433b-eaa0-4145-8859-e4b938e71551.jpg) I would also allow maximum one per day.

```
{
  ...
  "dateAdded": "2021-06-30",
  "dateUpdated": "2021-11-06T02:38:17.299560",
  "dateConsulted": [
    "2021-05-23",
    "2021-06-11",
    "2021-06-30",
  ],
  "dateCooked": [
    "2021-05-25",
    "2021-06-30",
  ],
  ...
}
```

From those data, it would then be possible to create new pages like "Top recipes" and add different filters for annual, monthly and all-time results. 
![image](https://user-images.githubusercontent.com/38195308/142893663-bf26cb0e-4af3-4386-b531-fc4a033aebf7.png)

Another idea of page could be something like "Rediscover recipes" where it would be possible to show different cards with recipes that might have been forgotten.
![image](https://user-images.githubusercontent.com/38195308/142902105-b95a9def-7c70-411f-a256-bf19349886f2.png)
For this one, I think that having a section in the settings would be great. Letting the user decide what kind of reminder he might want to have. (ex: Reminder if not cooked for 2 years and more, reminder if not consulted for 4 years, reminder if in top 10 of previous years and not cooked for 6 months, etc..)

**Pushing the idea a little bit further**
Mixing this idea with the possibilities of multiple users and groups, maybe having a "top recipes" for specific groups. Maybe allowing the users of the same groups to see the top recipes of other users, etc.

**Additional Information**
- [ ] If the feature is accepted I'm willing to submit a PR to provide this feature
> I would be willing to help conceptualize or test the solution, but I don't think I have the knowledge to help with coding.

**Comment by hay-kot** [comment]
Love this idea. Not something I'm going to be looking at in the near future, I need to stay focused on core functionality for now, I've been getting derailed by all these cool ideas! 

If anyone want's to jump on a Pull Request for this, please post and let me know.

**Reply by MatzeG1t** [reply]
Hi @hay-kot I have a suggestion regarding the "detailed by all these cool ideas" problem.

Right now I assume you look at the most upvoted features and core-features that are necessary from technical point of view.
It would be really helpful, if you would categorize the feature requests (aka cool ideas) with a user centred method.  This way you can strategically plan features.
You could for example use the KANO classification. The user facing basic features are always highest prio. Performance second and then improving reverse features and then researching on indifferent ones.
This would help to sprinkle one-two delight features into mealie from the backlog while always keeping focus on the basic things and technical maintenance/improvements. Though the latter could also be related to basic features. e.g. improving search duration.

Here are some examples I just made up to explain it quickly what KANO features of mealie could be:

Basic/MustHave: Features that just need to be there (e.g. edit ingredients), if they are missing/malfunction, its really putting user satisfaction down
Performance: Expected features, the better they are implemented, the better the user satisfaction
Delighters: unexpected features for the user, that create disproportionate satisfaction when present but cause no dissatisfaction when absent (e.g. these cooking stats in this feature request here, cook mode)
Indifferent: satisfy nor dissatisfy users regardless of their presence or implementation quality. These are a bit hard to spot and really differ depending on the Persona you are looking on.  e.g. for some people AI features might be that, for others kitchen utils, and so on
Reverse features: cause dissatisfaction when present. E.g. mandatory account creation for self hosted instances, where a Username might be enough.

Maybe you want to discuss that with the lead developers , if you want to elaborate on how to prioritize form the users point of view for the  mealie users  which are really grateful that you keep such a great product up n running 🙏

**Comment by ChipSkylark37** [comment]
I've been thinking of a request for this as a 'Statistics' page:

1. Highest Rated Recipes (Made Recently / Not Made Recently)
This could help you rediscover a recipe that was highly rated but not made recently
3. Lowest Rated Recipe (Made Recently / Not Made Recently)
Maybe you try the recipe again and alter it, maybe you decide its not for you and you want to get rid of it.
5. Most Made
6. Never Made
7. Want to Make (Not yet a feature) - Ive been using a 'Tag' for this and searching for that Tag of things that I tag as "Want to Make"
This would actually be a nice feature for the 'Randomize' feature - 'Try something I never made, but want to'
9. Most Time Required (Requires time to be counted and is not a current feature)

With most of these, the data is already present in the database and would just be a collection presented to the User via the GUI. As 'Recipe Finder' was recently added, perhaps this could be considered as well.

**Reply by parumpum** [reply]
"Want to Make" is a feature I was thinking about as well. My thought is a bookmark icon on the Recipe Card so that it can be easy to flag items as you scroll through. 

<img width="82" alt="image" src="https://github.com/user-attachments/assets/0ca3928f-a027-448a-a33f-c16912cf31d2" />

Anything regarding Time is difficult without a lot of parsing. OpenAI makes it simple to extract number of minutes, but not everyone has that enabled.

**Reply by ChipSkylark37** [reply]
A bookmark seems like a simple way to do it. Similar to the 'Heart / Favorite' but different use. Then an ability to filter on the 'bookmarked' recipes.

This would likely need to be able to be set from at least a few spots:
1. From the recipe search / card section
2. Inside the recipe page. 
- Would it make sense to have it only when 'editing' a recipe? I think even just from the main recipe page without having to edit is the cleanest.

**Reply by ChipSkylark37** [reply]
Thinking about placement:

If there's a potential the recipe image is itself red (using your bookmark example) it could be "lost". 

Would it make sense to be in a different location? So there's no ability for that to happen, in an area that has a white / black background area.

**Reply by parumpum** [reply]
I don't think the feature should require being in "edit" mode, similar to Favoriting.

The icon can be just about anywhere, including in the row along with the heart and stars. It is also possible to put a solid square behind it, so it would just be the small corner of the image that gets blocked out for the icon.

Placement can be discussed further, might need to go through some testing to see what feels good/natural. I think if we put it on the bottom, it may be best placed into the menu along with Add to Plan/List. I don't see it being visually pleasant next to the stars or the context menu dots.

**Reply by parumpum** [reply]
This is the sort of corner placement I envision

<img width="94" alt="image" src="https://github.com/user-attachments/assets/8211322e-c3fc-4dec-9cb7-b1222d48e7f3" />

Obviously just a quick mockup, color can be modified for light/dark modes and size can be dialed in to make sense.

**Reply by ChipSkylark37** [reply]
Top right feels like the best spot to me. Another version of a mockup via MSPaint:

![image](https://github.com/user-attachments/assets/f34295f7-52c0-4431-bb8b-ff9203305fed)

Plex has / had something similar, but they switched from the top corner to a little checkmark. I actually like the 'corner' version better, and that could change based on the mode / theme:

![image](https://github.com/user-attachments/assets/c139a02f-50ac-40f6-886b-da2037e80188).

Does this take into account _who_ bookmarked it? I know with multi-user mode, if one user set the stars, another member sees the stars greyed out as a visual indicator that it was rated, but not by me. I'm not sure if its similar for hearts. But if I (user 1) bookmarks something, would it bookmark for everyone, or just me? Perhaps it could also be visual:

- Red if it was bookmarked by the logged in user
- Grey if it was bookmarked by someone else? OR not at all? 

I personally would like to see if someone else bookmarked something, as maybe I'd make that dish for them. 

This opens a whole new conversation about seeing the 'owner' of the recipe as a little face in the recipe card and a much more robust method of filtering / searching / seeing things based on different users. But that is way outside of this 'bookmark' conversation.

**Reply by parumpum** [reply]
It'll be possible to use the existing framework provided by the recipe rating system to accomplish this.

Definitely need to share that a recipe is flagged to the household, as it may be different people flagging recipes and creating meal plans.

**Reply by ChipSkylark37** [reply]
I've been looking into some of the codebase recently but don't have my head wrapped around it enough to make changes. I'd love to be able to play around a bit on some of these, especially for referencing things already in the system and exposing them in other places (owner of recipe on recipe search card, yield of recipe in cook mode, etc).

**Reply by parumpum** [reply]
https://github.com/parumpum/mealie/tree/recipe-bookmarks

This branch adds the column to the database and an initial pass at a bookmark to play with just on the recipe card for now. Will need to see how it behaves between users.
